### PR TITLE
Lambda Workflows [Creates Lambda Functions] [3/5]

### PR DESCRIPTION
**Pull request by hsubbaraj-spiral**
## Describe your changes and why you are making these changes
Creates the Lambda Functions on users AWS account. Does the following:

1. Downloads the lambda images from Aqueduct Dockerhub
2. Creates ECR repos for the images and pushes to them.
3. Creates Lambda functions.   
## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/lib/engine/authenticate.go [hunk 1]
```diff
@@ -20,3 +27,29 @@ func AuthenticateK8sConfig(ctx context.Context, authConf auth.Config) error {
...
+
+	functionsToShip := [8]lambda_utils.LambdaFunctionType{
+		lambda_utils.FunctionExecutorType,
+		lambda_utils.ParamExecutorType,
+		lambda_utils.SystemMetricType,
+		lambda_utils.AthenaConnectorType,
+		lambda_utils.BigQueryConnectorType,
+		lambda_utils.PostgresConnectorType,
+		lambda_utils.S3ConnectorType,
+		lambda_utils.SnowflakeConnectorType,
```
Comment (hsubbaraj-spiral) [inline src/golang/lib/engine/authenticate.go:45]: Add MySql and SqlServer here.

**Review by likawind — COMMENTED** [review]
Curious if user need to provide k8s config when setting up lambda integration?

File: src/golang/lib/lambda/utils.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+	if err != nil {
+		return errors.Wrap(err, "Unable to map function type to image.")
+	}
+	//TODO: REPLACE LATEST WITH IMAGE VERSION NUMBER
+	imageVersionNumber := "latest"
+
+	stdout := &bytes.Buffer{}
+	stderr := &bytes.Buffer{}
+
+	cmd := exec.Command("docker", "pull", fmt.Sprintf("%s:%s", lambdaImageUri, imageVersionNumber))
```
Comment (saurav-c) [inline src/golang/lib/lambda/utils.go:44]: How does this image version number compare to the ones that are found in the constants file?

File: src/golang/lib/lambda/utils.go [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+			if aerr.Code() != ecr.ErrCodeRepositoryAlreadyExistsException {
+				log.Info(err)
+				return errors.Wrap(err, "Unable to create ECR repository.")
+			}
+		} else {
+			log.Info(err)
+			return errors.Wrap(err, "Unable to create ECR repository.")
+		}
+	}
+	log.Info(result)
```
Comment (saurav-c) [inline src/golang/lib/lambda/utils.go:89]: nit: reminder to rm debug logs in this file

File: src/golang/lib/lambda/utils.go (hunk 2 above)
Comment (hsubbaraj-spiral) [reply src/golang/lib/lambda/utils.go:44]: it should be the same value, I was unsure whether it was easier to set it as a constant or programmatically set it. Seemed easier if it was just one value we needed to change when doing a release.

**Comment by hsubbaraj-spiral** [comment]
> Curious if user need to provide k8s config when setting up lambda integration?

No they don't need to, k8s and lambda are entirely different integrations. Was there some code that was confusing about this?

**Pull request merged**